src/figuresClasses.ts

```diff
@@ -1,19 +1,92 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+  checkInitialData(): void;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.checkInitialData();
+  }
+
+  checkInitialData(): void {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('All sides should be > than zero!');
+    }
+
+    const sides: number[] = [this.a, this.b, this.c]
+      .sort((n: number, m: number) => m - n);
+
+    if (sides[0] >= sides[1] + sides[2]) {
+      throw new Error(
+        'The longes side should be < then the sum of two others!',
+      );
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
 
+    return Math.round(Math.sqrt(
+      s * (s - this.a) * (s - this.b) * (s - this.c),
+    ) * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: Shape = 'circle';
 
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    this.checkInitialData();
+  }
+
+  checkInitialData(): void {
```

Comment:
you can check the parameters inside the constructor without creating a new method

Reply:
for sure I can) but I wanted to make the constructor more compact)